payments controller and the interface

michaelt16 opened the pull request.

michaelt16 committed:
payments controller and the interface

dingdongg commented on src/controllers/PaymentsController.ts:
```diff
@@ -0,0 +1,31 @@
+import {Payments, IPayments} from "../database/models/Payments";
+
+export default class PaymentsContrtoller {
```
typo, should be "Controller"

dingdongg commented on src/database/models/Payments.ts:
```diff
@@ -0,0 +1,30 @@
+    Rejected
+} 
+
+export interface IPayments {
```
I think we should make the name non-plural. The name implies that this object holds a list of payments when the object only represents a single payment.

dingdongg commented on src/database/models/Payments.ts:
```diff
@@ -0,0 +1,30 @@
+export interface IPayments {
+
+    memberId: Schema.Types.ObjectId,
+    payment:string,
```
what does `payment` represent?

michaelt16 replied:
it was a stub, i edited it now.

dingdongg commented on src/database/models/Payments.ts:
```diff
@@ -0,0 +1,30 @@
+
+
+
+enum Status {
```
we can create a separate `enums` folder and create a file for the `Status` enum, that way we can reduce cluttering in the files

dingdongg replied:
when you do that, you also probably want to create functions that can convert from enum to string (and vice versa) so we can easily store on the DB

michaelt16 replied:
yup toString() works, and i created a seperate enum folder as well

dingdongg commented on src/services/Schedueler.ts:
```diff
@@ -0,0 +1,2 @@
+
```
typo in file name

dingdongg reviewed (changes requested):
made some comments

michaelt16 committed:
tested schema for payments

michaelt16 committed:
working on cron

michaelt16 committed:
finished cron and payment controllers

michaelt16 committed:
finished the hisotry function in ascending order and hisotry for a specific user

michaelt16 commented:
New and updated pull request.

- Added cron functionality called cronPayment() under UserController.ts

- Added two functions that builds payment history under PaymentController.ts, the createPaymentHistory and findUserHistory(userId)

- Updated schema so that it has the memberId paying for the payment.

I will make tests for these functions once its merged using mocha,chai

dingdongg commented on src/database/models/Payments.ts:
```diff
@@ -1,18 +1,16 @@
-
 const paymentsSchema = new Schema <IPayment>({
-    memberId: { type: Schema.Types.ObjectId, required: false},
+    memberId: { type: Schema.Types.ObjectId, required: true },
```
`
memberId: { type: Schema.Types.ObjectId, ref: "User" },
`

dingdongg reviewed (approved):
lgtm `2979abc73a110ac775ede2cae01d644472a3a229`

michaelt16 committed:
pr changes

michaelt16 committed:
Merge branch 'm3-backend' into m3-payments-controller
